api/src/main/java/uk/gov/hmcts/payment/api/controllers/FeesController.java

```diff
@@ -32,7 +36,11 @@ public FeesController(FeesService feesService) {
         @ApiResponse(responseCode = "400", description = "Fees not found")
     })
     @DeleteMapping(value = "/fees/{feeId}")
-    public ResponseEntity<Boolean> deleteFee(@PathVariable("feeId") String feeId) throws EmptyResultDataAccessException {
+    public ResponseEntity<Boolean> deleteFee(@NotNull @PathVariable("feeId") String feeId) throws EmptyResultDataAccessException {
+        Optional<PaymentFee> paymentFee = feesService.getPaymentFee(Integer.parseInt(feeId));
+        if (paymentFee.isEmpty()) {
+            throw new EmptyResultDataAccessException("Fee not found", 1);
+        }
```

Comment:
In my opinion, this check should go into service layer.

And a philosophical comment - what is the end goal of delete operation? to make sure that item doesn't exist. So if it doesn't exist, I would say operation was successful :)